.github/workflows/coral-pr-pipeline.yml

```diff
@@ -0,0 +1,95 @@
+name: Coral PR Pipeline
+
+on:
+  pull_request:
+    branches:
+      - main
+      - coral
+    paths:
+      - 'coral/**'
+      - '.github/workflows/coral-checks.yml'
+
+env:
+  NODE_VERSION: 16
+  PNPM_VERSION: 7.13.4
+  
+jobs:
+  static_code_analysis:
+    name: Static code analysis
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
```

Comment:
Please add the following to avoid the checkout action to write the github token to disk:

```
with:
    persist-credentials: false
```